tgui/packages/tgui/interfaces/Stack.js

```diff
@@ -0,0 +1,230 @@
+import { createSearch } from 'common/string';
+import { Fragment } from 'inferno';
+import { sortBy } from 'common/collections';
+import { useBackend, useLocalState } from "../backend";
+import { Box, Button, Input, NoticeBox, Section, Collapsible, Table } from "../components";
+import { Window } from "../layouts";
+
+export const Stack = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const {
+    amount,
+    recipes = [],
+  } = data;
+
+  const [
+    searchText,
+    setSearchText,
+  ] = useLocalState(context, 'searchText', '');
+
+  const testSearch = createSearch(searchText, item => {
+    return item.title;
+  });
+
+  const filterRecipes = (recipes, searchText) => {
+    return recipes
+      .filter(recipe => recipe.title !== undefined)
+      .map(recipe => recipe.sub_recipes 
+        ? { 
+          "title": recipe.title, 
+          "sub_recipes": filterRecipes(recipe.sub_recipes, searchText),
+        } 
+        : recipe)
+      .filter(recipe => recipe.sub_recipes 
+        ? recipe.sub_recipes.length > 0 
+        : testSearch(recipe));
+  };
+
+  const doSearch = searchText.length > 0;
+
+  const items = doSearch
+   && filterRecipes(recipes, searchText)
+    || recipes;
+
+  const height = Math.max(94 + recipes.length * 26, 250);
+
+  return (
+    <Window
+      width={400}
+      height={Math.min(height, 500)}
+      resizable>
```

Comment:
you can get rid of resizable, all tgui windows are automatically now

Reply:
Good catch